x/signutil/sign.go

```diff
@@ -0,0 +1,124 @@
+package signutil
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+)
+
+type MsgArbitrarySignature struct {
+	Message string         `json:"message"`
+	Signer  sdk.AccAddress `json:"signer"`
+}
+
+func (m MsgArbitrarySignature) Route() string {
+	return ModuleName
+}
+
+func (m MsgArbitrarySignature) Type() string {
+	return "arbitrary_signature"
+}
+
+func (m MsgArbitrarySignature) ValidateBasic() error {
+	if len(m.Message) == 0 {
+		return fmt.Errorf("empty msg")
+	}
+	if m.Signer.Empty() {
+		return fmt.Errorf("missing signer")
+	}
+	return nil
+}
+
+// GetSignBytes implements sdk.Message
+func (m MsgArbitrarySignature) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
+}
+
+// GetSigners implements sdk.Message
+func (m MsgArbitrarySignature) GetSigners() []sdk.AccAddress { return []sdk.AccAddress{m.Signer} }
+
+// getTxCmd clubs together all the CLI tx commands
+func getTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	configTxCmd := &cobra.Command{
+		Use:                        storeKey,
+		Short:                      fmt.Sprintf("%s transactions subcommands", storeKey),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	configTxCmd.AddCommand(flags.PostCommands(
+		sign(cdc),
+		verify(cdc),
+	)...)
+	return configTxCmd
+}
+
+func sign(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sign",
```

Comment:
Benjamin noted that this command shouldn't be called "sign" because it produces and unsigned message.  Let's rename it to "create".